app/services/db/monthly_sp_auth_count/total_monthly_auth_counts.rb

```diff
@@ -1,8 +1,27 @@
 module Db
   module MonthlySpAuthCount
     class TotalMonthlyAuthCounts
+      # @return [Array<Hash>]
       def self.call
-        sql = <<-SQL
+        oldest = ::SpReturnLog.first&.requested_at&.to_date&.beginning_of_month
+        newest = ::SpReturnLog.last&.requested_at&.to_date&.end_of_month
+
+        return [] if !oldest || !newest
+
+        Reports::MonthHelper.months(oldest..newest).flat_map do |month_range|
+          query_month(month_range)
```

Comment:
we *could* add a `with_retries` here to catch exceptions and retry these smaller queries. It would be speculative to add it, so I don't know if it's worth it to add now.

Reply:
update: adding `::date` casting in baa035c8205b9956cd4ce92ea0f2bee15a0754ed took each query from 236,259.6ms down to 17,160.4ms so I think this is much lower risk of being cancelled now, in the spirit of KISS, will leave out retrying for now